src/ir/LocalGraph.cpp

```diff
@@ -444,6 +440,73 @@ struct LocalGraphFlower
       }
     }
   }
+
+  // Given a bunch of gets, see if any of them reach the given set despite the
+  // obstacle expression stopping the flow whenever it is reached.
+  LocalGraphBase::SetInfluences
+  getSetInfluencesGivenObstacle(const LocalGraphBase::SetInfluences& gets,
```

Comment:
This all seems very similar to code that probably already exists. Can we use a bool template parameter and `if constexpr` to have a single implementation that can be configured to consider obstacles or not?